app/controllers/v1/memberships_controller.rb

```diff
@@ -6,7 +6,7 @@ def create
     @membership.user = current_user
     authorize @membership
     if @membership.save
-      render :show, status: :created
+      render 'v1/leaderboards/show', status: :created
```

Comment:
I'm not sure about rendering a `leaderboard` as a response from the `memberships#create`... Shouldn't we return the `membership` itself, and the FE would be in charge to redirecting to the correct `leaderboard` page based on the `leaderboard_id` of the membership?

Reply:
i suppose but a membership is just a join table. i dont think we have anything about memberships in the FE. We would have to make another API request to fetch the leaderboard, but I guess we're doing that no matter what.

updated.

Reply:
Yes, since after creating, we're redirecting to the Leaderboards page, the fetch happens no matter what.